drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,647 @@
+#include <assert.h>
```

Comment:
You will also need to add the copyright header here. You can copy it from any RIOT C file and adjust it as needed. Don't forget to add your name to the author list 😉